crates/solana-gateway-task-processor/src/component/message_payload.rs

```diff
@@ -0,0 +1,251 @@
+//! Message payload management module for Axelar Gateway integration.
+//!
+//! This module provides functionality to handle message payloads in the Solana blockchain,
+//! including initialization, writing, committing, and closing of message payload accounts.
+
+use axelar_solana_encoding::types::messages::Message;
+use axelar_solana_gateway::state::incoming_message::command_id;
+use eyre::Context as _;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt as _;
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::Keypair;
+use solana_sdk::signer::Signer as _;
+
+use super::send_transaction;
```

Comment:
You should use `send_gateway_tx` instead.
This is important for idempotency purposes, as it will check the GatewayError to figure out if we have a hard error or just a task that had already been executed.

Reply:
https://github.com/eigerco/axelar-solana-relayer/issues/12 I created a new issue for this